mirascope/core/gemini/_utils/_setup_call.py

```diff
@@ -53,7 +58,8 @@ def setup_call(
     elif extract:
         assert tool_types, "At least one tool must be provided for extraction."
         tool_config = call_kwargs.get("tool_config", {})
-        tool_config["function_calling_config"] = {"mode": "auto"}
+        # Ignore type checking, because gemini tool config is not yet typed correctly
+        tool_config["function_calling_config"] = {"mode": "auto"}  # pyright: ignore [reportGeneralTypeIssues, reportIndexIssue]
```

Comment:
ok so the solution here is the following (because Gemini is super annoying):

```python
assert tool_types, "At least one tool must be provided for extraction."
call_kwargs.pop("tool_config", None)
tool_config = ToolConfigDict()
tool_config.function_calling_config = {
    "mode": "any",
    "allowed_function_names": [tool_types[0]._name()],
}
call_kwargs["tool_config"] = tool_config
```

You'll need to import `from google.generativeai.types.content_types import ToolConfigDict`